release/v1alpha1/types.go

```diff
@@ -0,0 +1,76 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PromotionTestResults encapsulates the inputs to produce a new set of promotion tests and
+// the status of the promotion tests
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type PromotionTestResults struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec the inputs used to create the promotion test
+	Spec PromotionTestSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+
+	// Status is the current status of the promotion test
+	Status PromotionTestResultsStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// PromotionTestSpec has the information to represent a PromotionTest
+type PromotionTestSpec struct {
+	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+}
+
+// PromotionTestResultsStatus the status of all the promotion test jobs
+type PromotionTestResultsStatus struct {
+	JobResults []JobResults `json:"results,omitempty" protobuf:"bytes,1,rep,name=results"`
```

Comment:
As I recall, we have several different kinds of jobs.  Informing, Blocking, Analysis, Periodic.  I expected an array of each.

Reply:
Tell me what will write these.  We can have many readers, but only a single writer.

Reply:
Added entries for Blocking, Informing, and Analysis results.  AFAIK, the release-controller no longer deals in Periodic jobs (https://github.com/openshift/release/pull/20142)

Reply:
Currently, the release-controller is responsible for monitoring the status of the release verification jobs that it's responsible for.  It is not however monitoring the analysis jobs, but we can change that.